FreeRTOS/Test/Target/boards/pico/tests/smp/multiple_tasks_running/main.c

```diff
@@ -0,0 +1,146 @@
+ *
+ * https://www.FreeRTOS.org
+ * https://github.com/FreeRTOS
+ *
+ */
+
+/**
+ * @file main.c
+ * @brief The implementation of main function to start test runner task.
+ *
+ * Procedure:
+ *   - Initialize environment
+ *   - Run the test case
+ */
+
+/* Kernel includes. */
+#include "FreeRTOS.h" /* Must come first. */
+#include "task.h"     /* RTOS task related API prototypes. */
+
+#include "unity.h"    /* unit testing support functions */
+
+#include "pico/multicore.h"
+#include "pico/stdlib.h"
+
+/*-----------------------------------------------------------*/
+
+#define TASK_TESTRUNNER_PRIORITY    ( tskIDLE_PRIORITY + 1 )
+
+/*-----------------------------------------------------------*/
+
+static void prvTestRunnerTask( void * pvParameters );
+extern void vRunMultipleTasksRunningTest( void );
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief A start entry for unity to start with.
+ *
+ * @param[in] pvParameters parameter for task entry, useless in this test.
+ */
+static void prvTestRunnerTask( void * pvParameters )
+{
+    ( void ) pvParameters;
+
+    /* Execute test case provided in test.c */
+    vRunMultipleTasksRunningTest();
+
+    for( ; ; )
+    {
+        vTaskDelay( pdMS_TO_TICKS( 1000 ) );
+    }
+}
+/*-----------------------------------------------------------*/
+
+void vPortInitTestEnvironment( void )
+{
+    /* Setup LED I/O */
+    gpio_init( PICO_DEFAULT_LED_PIN );
+    gpio_set_dir( PICO_DEFAULT_LED_PIN, GPIO_OUT );
+    gpio_set_irq_enabled( PICO_DEFAULT_LED_PIN,
+                          GPIO_IRQ_LEVEL_LOW |
+                          GPIO_IRQ_LEVEL_HIGH |
+                          GPIO_IRQ_EDGE_FALL |
+                          GPIO_IRQ_EDGE_RISE,
+                          false );
+
+    /* Want to be able to printf */
+    stdio_init_all();
+
+    while( !stdio_usb_connected() )
+    {
+        gpio_put( PICO_DEFAULT_LED_PIN, 1 );
+        sleep_ms( 250 );
+        gpio_put( PICO_DEFAULT_LED_PIN, 0 );
+        sleep_ms( 250 );
+    }
+}
+/*-----------------------------------------------------------*/
+
+void vApplicationStackOverflowHook( TaskHandle_t xTask,
+                                    char * pcTaskName )
+{
+    ( void ) pcTaskName;
+    ( void ) xTask;
+
+    printf( "ERROR: Stack Overflow\n\0" );
+
+    /* Run time stack overflow checking is performed if
+     * configconfigCHECK_FOR_STACK_OVERFLOW is defined to 1 or 2.  This hook
+     * function is called if a stack overflow is detected.  pxCurrentTCB can be
+     * inspected in the debugger if the task name passed into this function is
+     * corrupt. */
+    for( ; ; )
+    {
+    }
+}
+/*-----------------------------------------------------------*/
+
+void vApplicationTickHook( void )
+{
+    static uint32_t ulCount = 0;
+
+    ulCount++;
+}
+/*-----------------------------------------------------------*/
+
+void vApplicationMallocFailedHook( void )
+{
+    printf( "ERROR: Malloc Failed\n\0" );

```

Comment:
Add an infinite loop here like vApplicationStackOverflowHook so that we can find the call stack when we break in debugger.